### PR TITLE
OSOE-545: Automatically assign GitHub issue corresponding to a pull requests based on Jira issue key in Lombiq.GitHub.Actions

### LINKED ISSUE #189: Automatically assign GitHub issue corresponding to a pull requests based on Jira issue key (OSOE-545)
Similar to https://github.com/Lombiq/GitHub-Actions/issues/151, let's also assign the GitHub issue corresponding to a PR to the author of the PR.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-545)

### PR DESCRIPTION
**Pull request by dorraj07**
[OSOE-545](https://lombiq.atlassian.net/browse/OSOE-545)
Fixes #189

[OSOE-545]: https://lombiq.atlassian.net/browse/OSOE-545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1 [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+else
+{
+    'No_Key_added'
+}
+
+$IssueQuery = "$Jira_Key in:title"
+$Output = gh issue list --search $IssueQuery --repo $GitHubRepository
+
+Write-Output $Output
```
Comment (wAsnk) [inline .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1]: Please delete it if it is not necessary.

File: .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1 [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+$FirstItem = ($Output | Select-Object -First 1)
+Write-Output $FirstItem
```
Comment (wAsnk) [inline .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1]: Please delete it if it is not necessary.

File: .github/workflows/validate-pull-request.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
         env:
           JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
+
```
Comment (wAsnk) [inline .github/workflows/validate-pull-request.yml]: Please remove this

File: .github/workflows/validate-submodule-pull-request.yml [hunk 4]
```diff
@@ -27,6 +27,11 @@ jobs:
           JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Assign issue to PR author
+        uses: Lombiq/GitHub-Actions/.github/actions/assign-issue-to-pr-author@issue/OSOE-545
```
Comment (wAsnk) [inline .github/workflows/validate-submodule-pull-request.yml]:
```suggestion
      - name: Assign issue to PR author
        if: github.event_name == 'pull_request' && github.event.action == 'opened'
        uses: Lombiq/GitHub-Actions/.github/actions/assign-issue-to-pr-author@issue/OSOE-545
```
I think this should be triggered only when opening the PR.

**Review by wAsnk — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-pull-request.yml [hunk 5]
```diff
@@ -20,7 +20,6 @@ jobs:
         env:
           JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
```
Comment (wAsnk) [inline .github/workflows/validate-pull-request.yml]: Please revert this change. This empty line is still needed.

**Review by wAsnk — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-pull-request.yml (hunk 3 above)
Comment (wAsnk) [inline .github/workflows/validate-pull-request.yml]: Okay, so what you did revert was actually the spaces which we don't need. What we need here is a simple new line, do you see this?
image
Those are spaces that are not needed. I pushed the fix for this so we are not doing more rounds on this.

**Review by wAsnk — APPROVED** [review]

**Comment by wAsnk** [comment]
Please don't forget to add "Fixes #no" to the description whenever you open a PR that fixes a GitHub issue. I added it now.

File: .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1 [hunk 6]
```diff
@@ -0,0 +1,28 @@
+param(
+    [string] $Body,
+    [string] $Assignee,
+    [string] $GitHubRepository
+)
+
+$Jira_Key = if ("$Body" -match '\[(.*?)\]')
```
Comment (Piedone) [inline .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1:7]: Shouldn't have this unconventional variable name and e.g. `$IssueQuery` below be caught by PS analysis, @BenedekFarkas @DAud-IcI?

Comment (sarahelsaig) [reply .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1:7]: [Here](https://github.com/PowerShell/PSScriptAnalyzer/tree/master/docs/Rules) are the list of built-in rules, I don't see anything regarding variable name styling.

We could create our own rule. If PSScriptAnalyzer can identify local variable declaration, that's when we should enforce no underscores and lower case first letter.

Comment (Piedone) [reply .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1:7]: I see, thanks.

Comment (BenedekFarkas) [reply .github/actions/assign-issue-to-pr-author/AssignIssueToPullRequestAuthor.ps1:7]: How about this (added some extra details): https://github.com/Lombiq/PowerShell-Analyzers/issues/31
